frontend/src/routes/Blogs.tsx

```diff
@@ -16,91 +17,87 @@ interface posts {
+      `${USER_API_ENDPOINT_PROD}/getUserInfo`,
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    );
+    return response.data.user;
+  };
+
+  const { data: userInfo, isError } = useQuery({
+    queryKey: ['userInfo', token],
+    queryFn: () => fetchUserInfo(token),
+    enabled: !!token,
+  });
 
-    try {
-      const response = await axios.get(
-        `${USER_API_ENDPOINT_PROD}/getUserInfo`,
-        {
-          headers: {
-            Authorization: `Bearer ${token}`,
-          },
-        }
-      );
-      console.log("Response", response);
-      console.log("Setting name", response.data.user.name);
-      setUserName(response.data.user.name);
-    } catch (error) {
-      toast.error("Error fetching user info");
+  useEffect(() => {
+    if (userInfo) {
+      setUserName(userInfo.name);
     }
+  }, [userInfo]);
+
+  const fetchPosts = async () => {
+    const response = await axios.get(`${BLOG_API_ENDPOINT_PROD}/bulk`, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+    return response.data.posts;
   };
 
+  const { data: posts, isLoading, isError : foo } = useQuery({
+    queryKey: ['posts', token],
+    queryFn: fetchPosts,
+    enabled: !!token,
+  });
+
+  useEffect(() => {
+    if (posts && posts.length === 0) {
+      setShowMessage(true);
+    }
+  }, [posts]);
+
   useEffect(() => {
-    fetchUserInfo(token);
     const handleStorageChange = () => {
       const newToken = localStorage.getItem("jwt-token");
       if (newToken !== token) {
         setToken(newToken);
-        fetchUserInfo(newToken); // Fetch new user info when token changes
       }
     };
     window.addEventListener("storage", handleStorageChange);
 
-    // Clean up event listener on component unmount
     return () => {
       window.removeEventListener("storage", handleStorageChange);
     };
   }, [token]);
 
-  useEffect(() => {
-    axios
-      .get(`${BLOG_API_ENDPOINT_PROD}/bulk`, {
-        headers: {
-          Authorization: `Bearer ${token}`,
-        },
-      })
-      .then((res) => {
-        console.log("print response ", res);
-        console.log(`Posts length:`, res.data.posts.length);
-        if (res.data.posts.length === 0 || res.data.posts) {
-          setShowMessage(true);
-          setPosts([]);
-        }
-        setPosts(res.data.posts);
-        const description = res.data.posts.map((post: { title: string }) => {
-          setDescription(post.title.substring(0, 5));
-        });
-        console.log("description is", description);
-        console.log("description2 is", description2);
-      })
-      .catch((err) => {
-        console.log("Error in get axios");
-        console.log(err);
-      });
-  }, [description2, token]);
-
   const truncateContent = (content: string, length: number) => {
     if (content.length <= length) return content;
     return content.substring(0, length) + "...";
   };
 
+  if(foo || isError) {
```

Comment:
Can you please change the name of this to some meaningful names.
like 

```bash
 if (postError || userError) 
```

Reply:
done